Framework/API/src/Algorithm.cpp

```diff
@@ -365,9 +367,7 @@ void Algorithm::cacheInputWorkspaceHistories() {
   auto cacheHistories = [this](const Workspace_sptr &ws) {
     if (auto group = dynamic_cast<const WorkspaceGroup *>(ws.get())) {
       m_inputWorkspaceHistories.reserve(m_inputWorkspaceHistories.size() + group->size());
-      for (const auto &memberWS : *group) {
-        m_inputWorkspaceHistories.emplace_back(memberWS);
-      }
+      std::copy(group->begin(), group->end(), std::back_inserter(m_inputWorkspaceHistories));
```

Comment:
```suggestion
      std::copy(group->cbegin(), group->cend(), std::back_inserter(m_inputWorkspaceHistories));
```

Reply:
Here, `group` is a `Mantid::API::WorkspaceGroup` that does not have `cbegin()` and `cend()` functions.